### ISSUE TITLE
Hdf5TreeModel: synchronize and remove do not work

### ISSUE THREAD

**Issue by kklmn** [silx-kit/silx#2503]
Hello,

I am using `silx.gui.hdf5.Hdf5TreeModel` in my app and now am trying to implement 'Synchronize' by invoking `synchronizeH5pyObject`. It has no effect. Then I tried to implement it myself and needed `removeH5pyObject` but the latter throws an error:

```
  File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\silx\gui\hdf5\Hdf5TreeModel.py", line 643, in removeH5pyObject
    self.removeIndex(qindex)
  File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\silx\gui\hdf5\Hdf5TreeModel.py", line 625, in removeIndex
    self._closeFileIfOwned(node)
  File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\silx\gui\hdf5\Hdf5TreeModel.py", line 573, in _closeFileIfOwned
    if f in obj:
  File "h5py\_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py\_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\h5py\_hl\group.py", line 335, in __contains__
    return self._e(name) in self.id
  File "C:\WPy-3701\python-3.7.0.amd64\lib\site-packages\h5py\_hl\base.py", line 132, in _e
    name = name.encode('ascii')
AttributeError: 'File' object has no attribute 'encode'
```
I would also try to reimplement it but the name mangling that you like so much makes it overly complex.

I was looking for an example implementation and have found 'silx/examples/hdf5widget.py'. Similarly there, if I load an hdf5 file, modify it externally and do 'synchronize', I see no effect. If I do 'remove', I get the same error above.

silx 0.10.0
h5py 2.8.0.post0

Thanks for looking into it!

**Comment by vasole** [comment]
Reproduced.

@vallsv 

The 'remove' seems to work if we change line 573 in HdfTreeModel.py from ``if f in obj:`` to ``if f.filename == obj.filename:``

**Comment by vallsv** [comment]
Yes the only sync function which work is `__synchronizeH5pyObject` from `silx.app.view.Viewer`. We unfortunately can't implement the synchronization the way it was designed on the model.

Sorry for the mangling, but that's the only way to make sure we can rework internally this functions. I seriously can't work without. If you want to touch something, it's better to copy-paste the class, or to do a dirty monkey patch (as always).

I will take a look.

**Comment by vallsv** [comment]
I think the line was supposed to be `f is obj`.
I will add a test.

**Comment by vallsv** [comment]
Then you can use `removeH5pyObject` if you like. This one is tested, then it should be fine.
We will fix `removeIndex` for the coming `0.10.1` release.
